SDWebImageAVIFCoder/Classes/SDImageAVIFCoder.m

```diff
@@ -13,41 +14,610 @@
+                                                         255,
+                                                         kvImageNoFlags);
+            if(err != kvImageNoError) {
+                free(argbPixels);
+                NSLog(@"Failed to convert to ARGB8888: %ld", err);
+                return;
             }
+            break;
+        }
+        case AVIF_PIXEL_FORMAT_YUV444:
+        {
+            err =
+            vImageConvert_YpCbCrToARGB_GenerateConversion(&matrix,
+                                                          &pixelRange,
+                                                          &convInfo,
+                                                          kvImage444CrYpCb8,
+                                                          kvImageARGB8888,
+                                                          kvImageNoFlags);
+            if(err != kvImageNoError) {
+                free(argbPixels);
+                NSLog(@"Failed to setup conversion: %ld", err);
+                return;
+            }
+
+            vImage_Buffer tmpBuffer = {
+                .data = calloc(avif->width * avif->height * 3, sizeof(uint8_t)),
+                .width = avif->width,
+                .height = avif->height,
+                .rowBytes = avif->width * 3,
+            };
+            if(!tmpBuffer.data) {
+                free(argbPixels);
+                return;
+            }
+            err = vImageConvert_Planar8toRGB888(&origCr, &origY, &origCb, &tmpBuffer, kvImageNoFlags);
+            if(err != kvImageNoError) {
+                NSLog(@"Failed to composite kvImage444CrYpCb8: %ld", err);
+                free(argbPixels);
+                free(tmpBuffer.data);
+                return;
+            }
+            vImageConvert_444CrYpCb8ToARGB8888(&tmpBuffer,
+                                               &argbBuffer,
+                                               &convInfo,
+                                               permuteMap,
+                                               255,
+                                               kvImageNoFlags);
+            free(tmpBuffer.data);
+            if(err != kvImageNoError) {
+                free(argbPixels);
+                NSLog(@"Failed to convert to ARGB8888: %ld", err);
+                return;
+            }
+            break;
+        }
+        case AVIF_PIXEL_FORMAT_YUV422:
+        {
+            err =
+            vImageConvert_YpCbCrToARGB_GenerateConversion(&matrix,
+                                                          &pixelRange,
+                                                          &convInfo,
+                                                          kvImage422YpCbYpCr8,
+                                                          kvImageARGB8888,
+                                                          kvImageNoFlags);
+            if(err != kvImageNoError) {
+                NSLog(@"Failed to setup conversion: %ld", err);
```

Comment:
When using Xcode Static Analyzer, it report the potential memory clean due to the clean up code. Can you have a check ? You can run the analyzer by Shift + Command + B as well.

`argbPixels` is leaked when this err occured.

There are also one in line 486 (leak `aYpCbCrBuffer`)





Some other smaller analyzer result (such as that `byteOrderNormal` variable is unused, you can delete), if you can , you can help me to solve them at all, or I solve them after this PR been merged.

Reply:
<img width="1286" alt="截屏2020-01-23下午1 55 03" src="https://user-images.githubusercontent.com/6919743/72960427-a2b99600-3de8-11ea-9d02-8e0e086779ba.png">

Reply:
I can't do that...analyzer reports some errors on libaom, and it does not analyze our sources:

<img width="1162" alt="スクリーンショット 2020-01-23 18 03 47" src="https://user-images.githubusercontent.com/51456946/72970696-8549f380-3e0b-11ea-9755-46358fd376a3.png">

How do you analyze it?
Are there way to avoid analyzing libaom, dav1d and rav1e?

Reply:
Hi, I can analyze the code (I opened Package.swift, instead of Examples/SDWebImageAVIFCoder.xcworkspace).

I suppressed all warnings in decoding functions:

<img width="1096" alt="スクリーンショット 2020-01-23 18 49 31" src="https://user-images.githubusercontent.com/51456946/72973719-1b344d00-3e11-11ea-846b-fbe5cdc75a08.png">